app/[locale]/dashboard/_components/staking-rewards.tsx

```diff
@@ -2,18 +2,20 @@
 import Image from 'next/image';
 import { useRouter } from 'next/navigation';
 import { useLocale, useTranslations } from 'next-intl';
+import { useContext } from 'react';
 import { Text, Title } from '@/components/typography';
 import { Box } from '@/components/ui/boxes';
 import { Button } from '@/components/ui/button';
-import { RewardsCalculatorModal } from '@/components/ui/modals';
+import { ButtonWithIcon } from '@/components/ui/button-with-icon';
+import { ModalContext } from '@/context';
 import { useTokenInfo } from '@/hooks/use-token-info';
 
 export default function StakingRewardsBlock () {
   const t  = useTranslations('Dashboard');
   const locale = useLocale();
   const router = useRouter();
   const { data: tokenInfo } = useTokenInfo();
-
+  const { openRewardsCalculatorModal } = useContext(ModalContext);
```

Comment:
You need to export something like this from ModalContext component

```tsx
export const useModal = () => {
  const context = useContext(ModalContext);
  if (context === undefined) {
    throw new Error('useModal must be used within a ModalProvider');
  }
  return context;
};
```

Reply:
And then just use it here as

```tsx
const { openRewardsCalculatorModal } = useModal();
```

Reply:
It will prevent you from usage context out of provider and will make code a bit cleaner